ssl/s3_srvr.c

```diff
@@ -2903,8 +2919,11 @@ int ssl3_get_cert_verify(SSL *s)
      * If key is GOST and n is exactly 64, it is bare signature without
      * length field
      */
-    if (n == 64 && pkey->type == NID_id_GostR3410_2001) {
-        len = 64;
+    if (n == 64 && (pkey->type == NID_id_GostR3410_2001 ||
+                    pkey->type == NID_id_GostR3410_2012_256)) {
+         len = 64;
+    } else if (n == 128 && (pkey->type == NID_id_GostR3410_2012_512)) {
+         len = 128;
```

Comment:
On Mon, Sep 21, 2015 at 10:42 PM, mattcaswell <notifications@github.com>
wrote:

> In ssl/s3_srvr.c
> <https://github.com/openssl/openssl/pull/407#discussion_r40015849>:
>
> > @@ -2903,8 +2919,11 @@ int ssl3_get_cert_verify(SSL *s)
> >       * If key is GOST and n is exactly 64, it is bare signature without
> >       * length field
> >       */
> > -    if (n == 64 && pkey->type == NID_id_GostR3410_2001) {
> > -        len = 64;
> > +    if (n == 64 && (pkey->type == NID_id_GostR3410_2001 ||
> > +                    pkey->type == NID_id_GostR3410_2012_256)) {
> > +         len = 64;
> > +    } else if (n == 128 && (pkey->type == NID_id_GostR3410_2012_512)) {
> > +         len = 128;
>
> Please check this workaround is actually needed for GOST12
>
>
>
I asked my former colleagues to make extra checks.I will report as soon as
I get the result.

-- 
SY, Dmitry Belyavsky